api/src/main/java/org/openmrs/module/queue/api/dao/impl/QueueEntryDaoImpl.java

```diff
@@ -41,84 +27,49 @@ public QueueEntryDaoImpl(@Qualifier("sessionFactory") SessionFactory sessionFact
 		super(sessionFactory);
 	}
 	
-	/**
-	 * @see QueueEntryDao#SearchQueueEntriesByConceptStatus(String, ConceptNameType, boolean, boolean)
-	 */
 	@Override
-	public Collection<QueueEntry> SearchQueueEntriesByConceptStatus(@NotNull String status, ConceptNameType conceptNameType,
-	        boolean localePreferred, boolean includeVoided) {
-		Criteria criteria = getCurrentSession().createCriteria(QueueEntry.class, "qe");
-		//Include/exclude retired queues
-		includeVoidedObjects(criteria, includeVoided);
-		criteria.add(
-		    Property.forName("qe.status").in(conceptByNameDetachedCriteria(status, localePreferred, conceptNameType)));
-		
+	public List<QueueEntry> getQueueEntries(QueueEntrySearchCriteria searchCriteria) {
+		Criteria criteria = createCriteriaFromSearchCriteria(searchCriteria);
 		return criteria.list();
 	}
 	
-	/**
-	 * @see org.openmrs.module.queue.api.dao.QueueEntryDao#getQueueEntriesCountByConceptStatus(String,
-	 *      ConceptNameType, boolean)
-	 */
 	@Override
-	public Long getQueueEntriesCountByConceptStatus(@NotNull String conceptStatus, ConceptNameType conceptNameType,
-	        boolean localePreferred) {
-		Criteria criteria = getCurrentSession().createCriteria(QueueEntry.class, "qe");
-		//Include/exclude retired queues
-		includeVoidedObjects(criteria, false);
-		criteria.add(Restrictions.and(Restrictions.isNull("qe.endedAt"), Restrictions.isNotNull("qe.startedAt")));
-		criteria.add(Property.forName("qe.status")
-		        .in(conceptByNameDetachedCriteria(conceptStatus, localePreferred, conceptNameType)));
+	public Long getCountOfQueueEntries(QueueEntrySearchCriteria searchCriteria) {
+		Criteria criteria = createCriteriaFromSearchCriteria(searchCriteria);
 		criteria.setProjection(Projections.rowCount());
-		
 		return (Long) criteria.uniqueResult();
 	}
 	
-	@Override
-	public String generateVisitQueueNumber(Location location, Queue queue) {
```

Comment:
I think we should retain this before we decide the way forward

Reply:
Thanks @CynthiaKamau - this functionality wasn't removed in this PR, it was just moved into entirely into the service.  So the existing visit queue number functionality remains as it was, just not using a custom DAO method to do it.